core/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -229,19 +229,22 @@ private void applyDeletedIndices(final ClusterChangedEvent event) {
                 logger.debug("[{}] cleaning index, no longer part of the metadata", index);
             }
             final IndexService idxService = indicesService.indexService(index);
-            final IndexSettings indexSettings;
+            IndexMetaData metaData;
             if (idxService != null) {
-                indexSettings = idxService.getIndexSettings();
+                final IndexSettings indexSettings = idxService.getIndexSettings();
                 deleteIndex(index, "index no longer part of the metadata");
+                sendDeletedEvent(event.state(), index, indexSettings, localNodeId);
+            } else if ((metaData = previousState.metaData().index(index)) != null) {
+                // The deleted index was part of the previous cluster state, but not loaded on the local node
+                final IndexSettings indexSettings = new IndexSettings(metaData, settings);
+                indicesService.deleteUnassignedIndex("deleted index was not assigned to local node", metaData, event.state());
```

Comment:
I think this message is misleading (it sounds close to I'm deleted this index because no shards are assigned to the local node). Maybe "index no longer part of the metadata (and no shards were assigned locally)"?